File: Semester3/ProgrammingParadigms/C1/2/Scala/main.scala
object app2 extends App {
    def count[T](value : T, list : List[T]):Int = {

    if (list != null && list.nonEmpty) {
      val isEqual = if (value == list.head) 1 else 0
      isEqual + count(value, list.tail)
    } else 0
  }


    var myList = List(1,2,1,2,3,4,2,7,8,3,6,2)

    var result = count(2, myList)

    println(result)
}

File: Semester3/ProgrammingParadigms/C1/2/OCaml/main.ml


let rec count(value, list) = (
    let isEqual = (fun (y, z) -> if y = z then 1 else 0) in
    if list = [] then 0
    else isEqual(value, List.hd list) + count(value, List.tl list)
);;


let list = [1;2;3;5;5;6;5;7;8;5;9;5;3;6;5];;

let result = count(5, list);;

Printf.printf "%d\n" result;